packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts

```diff
@@ -37,8 +39,52 @@ export function constructCasingsGenerator({
             let camelCaseName = camelCase(name);
             let pascalCaseName = upperFirst(camelCaseName);
             let snakeCaseName = snakeCase(name);
-            const camelCaseWords = words(camelCaseName);
-            if (smartCasing) {
+
+            // generatorsYml.CasingVersion.V0 should effectively be treated as smartCasing
+            if (
+                smartCasing ||
+                casingVersion === generatorsYml.CasingVersion.V0 ||
+                casingVersion === generatorsYml.CasingVersion.V1
+            ) {
```

Comment:
i would have thought that `v0` would not default to smart casing? 

Some context on the PR about what is `v0` and what is `v1` would be helpful